cli/src/main.rs

```diff
@@ -45,58 +76,118 @@ struct Connection {
 
 #[derive(Subcommand)]
 enum Commands {
-    Status,
-    Deploy(DeployArgs),
-}
+    // status
+    Status(status::StatusArgs), //aka topology
+
+    // deployment
+    DeployResource(deploy_resource::DeployResourceArgs),
+
+    // process instance
+    CreateProcessInstance(create_process_instance::CreateProcessInstanceArgs),
+    CancelProcessInstance(cancel_process_instance::CancelProcessInstanceArgs),
 
-#[derive(Args)]
-struct DeployArgs {
-    #[clap(required = true, value_parser, value_name = "FILE")]
-    resources: Vec<PathBuf>,
+    // message
+    PublishMessage(publish_message::PublishMessageArgs),
+
+    // incident
+    ResolveIncident(resolve_incident::ResolveIncidentArgs),
+
+    // variables
+    SetVariables(set_variables::SetVariablesArgs),
+
+    //jobs
+    ActivateJobs(activate_jobs::ActivateJobsArgs),
+    CompleteJob(complete_job::CompleteJobArgs),
+    FailJob(fail_job::FailJobArgs),
+    UpdateRetries(update_retries::UpdateRetriesArgs),
+    ThrowError(throw_error::ThrowErrorArgs),
 }
 
 impl From<Connection> for zeebe_client::Connection {
     fn from(conn: Connection) -> Self {
         match conn.address {
-            Some(addr) => zeebe_client::Connection::Address(addr),
-            None => zeebe_client::Connection::HostPort(conn.host, conn.port),
+            Some(addr) => zeebe_client::Connection::Address {
+                insecure: conn.insecure,
+                addr,
+            },
+            None => zeebe_client::Connection::HostPort {
+                insecure: conn.insecure,
+                host: conn.host,
+                port: conn.port,
+            },
         }
     }
 }
 
+impl TryFrom<Authentication> for zeebe_client::Authentication {
+    type Error = Report;
+
+    fn try_from(auth: Authentication) -> Result<zeebe_client::Authentication> {
+        match (auth.client_id, auth.client_secret) {
+            (None, None) => Ok(zeebe_client::Authentication::Unauthenticated),
+            (Some(client_id), Some(client_secret)) => Ok(zeebe_client::Authentication::Oauth2(
+                zeebe_client::auth::OAuth2Config {
+                    client_id,
+                    client_secret,
+                    auth_server: auth.authorization_server,
+                    audience: "zeebe.camunda.io".to_owned(),
+                },
+            )),
+            _ => Err(color_eyre::eyre::eyre!(
+                "Partial authentication info, needs at least client id and client secret"
+            )),
+        }
+    }
+}
+
+#[async_trait]
+trait ExecuteZeebeCommand {
+    type Output: Debug;
+    async fn execute(self, client: &mut ZeebeClient) -> Result<Self::Output>;
+}
+
+fn install_tracing() {
```

Comment:
:question: Do we want to move this into `client` crate?

Reply:
No, this setup is different for each environment. For production usage, I'd want to setup tracing with an OpenTelemetry exporter for example. For Cli we are setting it up so we get something human readable in stdout.